oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java

```diff
@@ -246,12 +248,49 @@ public <T> Predicate<T> getFilterPredicate(Set<IndexDefinition> indexDefinitions
     }
 
     /**
-     * @param pattern Pattern for a custom excludes regex based on which paths would be filtered out
+     * @param pattern              Pattern for a custom excludes regex based on which paths would be filtered out
      * @param typeToRepositoryPath Function to convert type <T> to valid repository path of type <String>
      * @param <T>
      * @return Return a predicate that should test true for all paths that do not match the provided regex pattern.
      */
     public <T> Predicate<T> getFilterPredicateBasedOnCustomRegex(Pattern pattern, Function<T, String> typeToRepositoryPath) {
         return t -> !pattern.matcher(typeToRepositoryPath.apply(t)).find();
     }
+
+    /**
+     * Computes the total size of the generated index data. This method is intended to be used when creating Lucene
+     * indexes, which are created locally. With Elastic, this will not include the Lucene files since the indexes
+     * are updated remotely.
+     *
+     * @return The total size of the index data generated or -1 if there is some error while computing the size.
+     */
+    public long computeSizeOfGeneratedIndexData() {
+        try {
+            File localIndexDir = getLocalIndexDir();
+            long totalSize = 0;
+            if (localIndexDir == null || !localIndexDir.isDirectory()) {
+                LOG.warn("Local index directory is invalid, this should not happen: {}", localIndexDir);
+                return -1;
+            } else {
+                // Each index is stored in a separate directory
+                File[] directories = localIndexDir.listFiles(File::isDirectory);
+                // Print the indexes in alphabetic order
+                Arrays.sort(directories);
+                StringBuilder sb = new StringBuilder();
+                for (File indexDir : directories) {
+                    long size = FileUtils.sizeOfDirectory(indexDir);
+                    totalSize += size;
+                    long numberOfFiles = indexDir.listFiles(File::isFile).length;
+                    sb.append("\n  - " + indexDir.getName() + ": " + numberOfFiles + " files, " +
+                            size + " (" + FileUtils.byteCountToDisplaySize(size) + ")");
+                }
+                LOG.info("Total size of index data generated: {} ({}){}",
+                        totalSize, FileUtils.byteCountToDisplaySize(totalSize), sb);
+                return totalSize;
+            }
+        } catch (Throwable t) {
+            LOG.warn("Error while computing size of generated index data: {}", t.toString());
```

Comment:
I would log the stack trace as well
```suggestion
            LOG.warn("Error while computing size of generated index data: {}", t.toString(), t);
```

Reply:
ok, changed.